test/app-tap/gh-2717-no-quit-sigint.test.lua

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env tarantool
+
+local tap = require('tap')
+local popen = require('popen')
+local process_timeout = require('process_timeout')
+local fio = require('fio')
+local clock = require('clock')
+
+--
+-- gh-2717: tarantool console quit on sigint
+--
+local file_read_timeout = 3.0
+local file_read_interval = 0.2
+local file_open_timeout = 3.0
+
+local TARANTOOL_PATH = arg[-1]
+local prompt = 'tarantool> '
+local test = tap.test('gh-2717')
+
+test:plan(3)
+
+local ph = popen.shell(TARANTOOL_PATH .. ' -i', 'r')
+assert(ph.pid, 'popen error while executing ' .. TARANTOOL_PATH)
+
+local start_time = clock.monotonic()
+local deadline = 10.0
+
+local output = ph:read()
+while output:find('tarantool>') == nil and clock.monotonic() - start_time < deadline do
+    output = output .. ph:read()
+end
```

Comment:
We read from popen handle without a timeout, so the deadline will not work if we're stuck.

Reply:
I see that you added a timeout, but not for all `:read()` calls.

Reply:
(And, please, prefer `{timeout = 5}` style: no spaces around curly brackets and spaces around equal sign.)

Reply:
Okay, all the `read()` calls have some timeout now, so it cant' be stuck. But I see there are some builds, where this test fails, I'm researching the problem currently.